labgrid/external/remote_tmpdir.py

```diff
@@ -0,0 +1,53 @@
+"""
+    RemoteTmpdir is a class that maintains a temporary directory on a
+    remote linux system.
+    Ex. when copying a test script or binary to DUT, RemoteTmpdir assists
+    to prevent pollution of the target.
+
+    RemoteTmpdir is to be created in a fixture, which can take care of a
+    testmodule in a subdirectory. Recommended scope is function, to avoid
+    polution between tests.
+
+    See examples/remote_tmpdir
+"""
+
+import os
+
+
+class RemoteTmpdir:
+
+    # scriptdir is a directory relative to pytest root
+    def __init__(self, shell, testdir=None):
+        stdout, stderr, returncode = shell.run("mktemp -d")
+        assert returncode == 0
+        assert len(stdout) == 1
+        # Use path to prefix copied shells
+        # shell.run(tmp.path+"helloworld")
+        self.path = stdout[0] + '/'
+        self.shell = shell
+        self.testdir = testdir
+
+        if self.testdir is not None and os.path.isfile(self.testdir):
+            raise Exception("RemoteTmpdir: {} is not a directory".format(
+                            self.testdir))
+
+    ## Copy a file or contents of directory the created tmdir
+    # putitem   file or directory to copy
+    #           does not create sub directories, but copies all files
+    def put(self, putitem):
+        # resolve relative paths
+        if not os.path.isabs(putitem) and self.testdir is not None:
+            putitem = os.path.join(self.testdir, putitem)
+
+        if os.path.isfile(putitem):
+            self.shell.put(putitem,
+                           self.path+os.path.basename(putitem))
+        else:
+            # then it is a whole directory to copy
+            for dirfile in os.listdir(putitem):
+                if os.path.isfile(os.path.join(putitem, dirfile)):
+                    self.shell.put(os.path.join(putitem, dirfile),
+                                   self.path+dirfile)
+
+    def get(self, getfile, localdir):
```

Comment:
This could also be extended to support multiple files:

```python
    def get(self, *files, localdir=None):
        for path in files:
            remotepath = self.path + path
            self.filetransfer.get(remotepath, localdir or self.basedir)
```